packages/moonstone/TimePicker/TimePickerBase.js

```diff
@@ -43,18 +43,21 @@ class HourPicker extends React.Component {
 		};
 	}
 
-	UNSAFE_componentWillReceiveProps (nextProps) {
-		const {children, value} = this.props;
-		const {children: nextChildren, value: nextValue} = nextProps;
+	static getDerivedStateFromProps (props, state) {
+		const hours = props.hasMeridiem ? hours12 : hours24;
 
-		this.setState({
-			noAnimation: children[value] === nextChildren[nextValue]
-		});
+		return {
```

Comment:
Let's add a condition to return `null` when there's no change in the value.

e.g.
```

		if (state.prevValue !== props.value) {
			return {
				noAnimation: hours[state.prevValue] === hours[props.value],
				prevValue: props.value
			};
		}
		return null;
```